crates/db_schema/src/impls/tagline.rs

```diff
@@ -0,0 +1,54 @@
+use crate::{
+  source::tagline::{Tagline, TaglineInsertForm, TaglineUpdateForm},
+  traits::Crud,
+};
+use diesel::{dsl::*, result::Error, PgConnection, QueryDsl, RunQueryDsl};
+
+impl Crud for Tagline {
+  type InsertForm = TaglineInsertForm;
+  type UpdateForm = TaglineUpdateForm;
+  type IdType = i32;
+  fn create(conn: &mut PgConnection, new_tagline: &TaglineInsertForm) -> Result<Self, Error> {
+    use crate::schema::tagline::dsl::*;
+    insert_into(tagline)
+      .values(new_tagline)
+      .get_result::<Self>(conn)
+  }
+
+  fn read(conn: &mut PgConnection, tagline_id: i32) -> Result<Self, Error> {
+    use crate::schema::tagline::dsl::*;
+    tagline.find(tagline_id).first::<Self>(conn)
+  }
+
+  fn update(
+    conn: &mut PgConnection,
+    tagline_id: i32,
+    new_tagline: &TaglineUpdateForm,
+  ) -> Result<Self, Error> {
+    use crate::schema::tagline::dsl::*;
+    diesel::update(tagline.find(tagline_id))
+      .set(new_tagline)
+      .get_result::<Self>(conn)
+  }
+
+  fn delete(conn: &mut PgConnection, tagline_id: i32) -> Result<usize, Error> {
+    use crate::schema::tagline::dsl::*;
+    diesel::delete(tagline.find(tagline_id)).execute(conn)
+  }
+}
+
+impl Tagline {
+  pub fn get_random(conn: &mut PgConnection) -> Option<Self> {
+    use crate::schema::tagline::dsl::*;
+    sql_function!(fn random() -> Integer);
+    match tagline.order(random()).first::<Self>(conn) {
```

Comment:
No need to convert this to an option here, none of the other DB calls do that for missing values. We can convert it to an option using `.ok()` when this is called.